### PR TITLE
Move GeoShapeIndexer#prepareForIndexing to its own utility class

### PR DESCRIPTION
**Pull request by iverase**
The method `GeoShapeIndexer#prepareForIndexing`  is just a utility that transforms geometries into a Lucene friendly format by normalising latitudes and longitudes and breaking geometries that crosses the dateline. In many cases we just want to perform such action so we need to create a dummy GeoShapeIndexer. Therefore this PR proposes to move that logic into its own utility class called `GeometryNormalizer`.

In addition GeoShapeIndexer takes a boolean to define the orientation which I found difficult to read. Therefore this PR changes the boolean to the Orientation enum in order to make it easier to read.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/geo/GeometryNormalizer.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.common.geo.GeoUtils.normalizePoint;
+
+/**
+ * Transforms provided {@link Geometry} into a lucene friendly format by normalizing latitude and longitude
+ * coordinates and breaking geometries that cross the dateline.
+ */
+public class GeometryNormalizer {
```
Comment (imotov) [inline server/src/main/java/org/elasticsearch/common/geo/GeometryNormalizer.java:33]: That looks like an utility class, why not make it `final` as well then?

**Review by imotov — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/docs

File: server/src/main/java/org/elasticsearch/common/geo/GeometryNormalizer.java (hunk 1 above)
Comment (iverase) [reply server/src/main/java/org/elasticsearch/common/geo/GeometryNormalizer.java]: makes sense

**Pull request merged**